File: src/common/nodejs/src/errors/not-authorized-error.ts
import { CustomError } from './custom-error';

export class NotAuthorizedError extends CustomError {
  statusCode = 401;

  constructor(public reason: string = 'Not Authorized') {
    super(reason);

    Object.setPrototypeOf(this, NotAuthorizedError.prototype);
  }

  serializeErrors() {
    return [{ message: this.reason }];
  }
}

File: src/common/django/shared/errors/not_authorized_error.py
from typing import List
from .custom_error import CustomError, IErrorStruct


class NotAuthorizedError(CustomError):
    status_code = 401

    def __init__(self, message: str):
        self.message = message
        super().__init__(self.message)

    def serialize_errors(self) -> List[IErrorStruct]:
        return [IErrorStruct(message=self.message, field=self.field).to_dict()]
